presto-hive/src/main/java/com/facebook/presto/hive/HiveCoercer.java

```diff
@@ -346,5 +468,15 @@ else if (i < rowBlock.getFieldCount()) {
             }
             return RowBlock.fromFieldBlocks(valueIsNull.length, Optional.of(valueIsNull), fields);
         }
+
+        @Override
+        public TupleDomainFilter toCoercingFilter(TupleDomainFilter filter)
+        {
+            if (filter == IS_NULL || filter == IS_NOT_NULL) {
```

Comment:
what about ALWAYS_FALSE and ALWAYS_TRUE

Reply:
@bhhari We should't have these. If there is an ALWAYS_FALSE filter, the optimizer should replace Scan with empty `Values`.